csharp/test/Ice/ami/TestI.cs

```diff
@@ -109,8 +109,14 @@ public void finishDispatch(Current current)
             }
         }
 
+        public int set(int value, Current current)
+        {
+            return Interlocked.Exchange(ref _value, value);
```

Comment:
There's no threading concern here since set dispatch are dispatched using an object adapter configured with serialized dispatch. I used this for the one-liner... I'll change it to do a "manual" exchange.